### PR TITLE
#187300153 Admin Account Disable and Activate

### PR DESCRIPTION
**Pull request by musabehonore**
# What does this PR do?
This PR implements the functionality for Admin to disable and activate any user account on the app.

# Description of Task to be completed?
An admin will be able to disable and activate any user's account and send them an email explaining the reasons for the suspension.
# How should this be manually tested?
### For testing only
npm install
npm run test
### For running the app
npm run install
npm run demigrate
npm run migrate
npm run seed
npm run dev

# Any background context you want to provide?
No background feature added.

# What are the relevant pivotal tracker/Trello stories?
-Story ID: #187300153

### CODE REVIEW

File: src/controllers/loginController.ts [hunk 1]
```diff
@@ -17,16 +17,20 @@ const loginController = async (req: Request, res: Response) => {
         'User not found or Invalid Credentials'
       );
     }
+
+    if (user.status !== 'Active') {
```
Comment (fab-ryan) [inline src/controllers/loginController.ts]: please use boolean not a string

File: src/controllers/userStatusController.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  res: Response
+) => {
+  try {
+    const { id } = req.params;
+    const { action, reasons } = req.body;
+    const user = await Database.User.findOne({ where: { id } });
+    if (!user) {
+      return sendResponse<null>(res, 404, null, 'User not found');
+    }
+    if (user.status === statuses[action]) {
```
Comment (fab-ryan) [inline src/controllers/userStatusController.ts]: you don't need actions can be vice versa true or false

File: src/database/migrations/20240411085039-add_auth_columns_to_users.js [hunk 3]
```diff
@@ -16,11 +16,17 @@ module.exports = {
       allowNull: false,
       defaultValue: false,
     });
+    await queryInterface.addColumn('users', 'status', {
+      type: Sequelize.STRING,
+      allowNull: false,
+      defaultValue: 'Active',
```
Comment (fab-ryan) [inline src/database/migrations/20240411085039-add_auth_columns_to_users.js]: please if you have new task that require to add new column please create it's own migration

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on those commets

File: src/routers/userRoute.ts [hunk 4]
```diff
@@ -45,5 +46,6 @@ router.patch(
 );
 
 router.get('/users/verify-email/:token', verifyEmail);
+router.post('/users/:id/accountStatus', isAuthenticated, isAdmin, userStatus);
```
Comment (fab-ryan) [inline src/routers/userRoute.ts]: we have naming conversation of api like if verifyEmail should be verify-email or account-status

**Review by fab-ryan — CHANGES_REQUESTED** [review]
change that endpoint naming

**Review by Mukunzijames — APPROVED** [review]

**Review by Sophie20202 — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**